File: category-frontend/src/services/orderService.js
import api from "../api/axios";

export const getOrders = (
    page = 0,
    size = 10,
    status = null,
    orderStatus = null,
    customerName = null,
    customerEmail = null
) => {
    const params = new URLSearchParams({ page, size });
    if (status !== null) params.append("status", status);
    if (orderStatus) params.append("orderStatus", orderStatus);
    if (customerName) params.append("customerName", customerName);
    if (customerEmail) params.append("customerEmail", customerEmail);
    return api.get(`/orders?${params.toString()}`);
};

export const getOrderById = (id) => api.get(`/orders/${id}`);

export const placeOrder = (data) => api.post("/orders", data);

export const updateOrderStatus = (id, data) => api.patch(`/orders/${id}/status`, data);

export const cancelOrder = (id) => api.patch(`/orders/${id}/cancel`);

File: category-backend/src/main/java/com/namit/categorybackend/order/service/OrderService.java
package com.namit.categorybackend.order.service;

import com.namit.categorybackend.common.response.PagedResponse;
import com.namit.categorybackend.order.dto.OrderResponse;
import com.namit.categorybackend.order.dto.OrderSummaryResponse;
import com.namit.categorybackend.order.dto.PlaceOrderRequest;
import com.namit.categorybackend.order.dto.UpdateOrderStatusRequest;
import com.namit.categorybackend.order.enums.OrderStatus;

public interface OrderService {
    OrderResponse placeOrder(PlaceOrderRequest request);
    PagedResponse<OrderSummaryResponse> getAllOrders(int page, int size, Boolean status, OrderStatus orderStatus, String customerName, String customerEmail);
    OrderResponse getOrderById(Long id);
    OrderResponse updateOrderStatus(Long id, UpdateOrderStatusRequest request);
    OrderResponse cancelOrder(Long id);
}
